### PR TITLE
Feature/ja 54 schedule item controller

### PR DESCRIPTION
**Pull request by skrawus**
Create schedule item controller

I created a controller and views for schedule-item. Merged develop into the branch and tested it. Names are left as they are at default so we can agree on the names later. There is also a new navigation link added in _Layout.cshtml only for tests.

### CODE REVIEW

File: TutorLizard.Web/Controllers/ScheduleItemController.cs [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        }
+
+        // GET: ScheduleItemController/Details/5
+        public ActionResult Details(int id)
+        {
+            var scheduleItem = _dataAccess.GetScheduleItemById(id);
+
+            if(scheduleItem == null)
+            {
+                return RedirectToAction("Error");
```
Comment (Zjyslav) [inline TutorLizard.Web/Controllers/ScheduleItemController.cs]: We don't have `Error` action in this controller. I get 404. Maybe let's redirect to Index.

File: TutorLizard.Web/Controllers/ScheduleItemController.cs [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        {
+            try
+            {
+                _dataAccess.DeleteScheduleItemById(id);
+
+                return RedirectToAction(nameof(Index));
+            }
+            catch
+            {
+                return View(_dataAccess.GetAllScheduleItems());
```
Comment (Zjyslav) [inline TutorLizard.Web/Controllers/ScheduleItemController.cs]: This view expects single ScheduleItem. You can simply give it `model`.

**Review by Zjyslav — COMMENTED** [review]
Aside from the two comments I had in code:
- Add the same check for null object as in `Details` to `Edit` and `Delete`
- Remove `Id` field from create View - it's not passed to `_dataAccess` anyway, so setting it creates false expectations.

**Review by Zjyslav — CHANGES_REQUESTED** [review]
My previous comments were supposed to be request changes.

**Comment by skrawus** [comment]
Ok, thanks. I've added the null checks but I'm wondering about this part of your comment:
"Remove `Id` field from create View". There is only AdId and I think we should pass it, as the `scheduleItemRequest` has to be attached to a specific ad

**Comment by Zjyslav** [comment]
> Ok, thanks. I've added the null checks but I'm wondering about this part of your comment: "Remove `Id` field from create View". There is only AdId and I think we should pass it, as the `scheduleItemRequest` has to be attached to a specific ad

We need `AdId`, we don't need `Id`.
Notice that when you process the creation, you call `_dataAccess.CreateScheduleItem(adId, dateTime);`, so if only these 2 field are used, why ask user to provide anything else?

**Comment by skrawus** [comment]
Got it! Corrected

**Review by Zjyslav — APPROVED** [review]
It's good enough and I don't want to hold it back, but you might want to remove `Id` also from Edit view, because even if I try to change it, `Id` stays the same.

**Review by Zjyslav — APPROVED** [review]

**Pull request merged**